src/artifacts.js

```diff
@@ -248,8 +329,42 @@ api.declare({
     }
 
     // Check that details match, unless this has storageType 'reference', we'll
-    // workers to overwrite redirect artifacts
-    if (storageType !== 'reference' &&
+    // workers to overwrite redirect artifacts.  We handle blob artifacts
+    // differently than other all the other artifacts.  We consider a second
+    // creation of the same blob a non-error.  If the blob is a multipart
+    // upload we need to handle an already-existing uploadId gracefully.  In
+    // the case of a conflict, we will cancel the uploadId created in *this*
+    // request and will eventually return requests based on the existing
+    // uploadId.
+    if (storageType === 'blob') {
+      let storedWithoutUploadId = _.omit(artifact.details, 'uploadId');
+      let inputWithoutUploadId = _.omit(details, 'uploadId');
+
+      // The two good conditions are that the details are identical or that
+      // they're multipart uploads and the uploadId is the only differing
+      // attribute.  In that case, we'll cancel the uploadId we just created
+      if (_.isEqual(artifact.details, details)) {
+        // Do nothing!
+      } else if (input.parts && _.isEqual(storedWithoutUploadId, inputWithoutUploadId)) {
+        if (artifact.details.uploadId !== details.uploadId) {
+          await this.s3Controller.abortMultipartUpload({
+            bucket: details.bucket,
+            key: details.key,
+            uploadId: details.uploadId,
+          });
```

Comment:
Don't we also have to set `details.uploadId = artifact.details.uploadId`? That would explain the error wrt. idempotency we saw in tests in travis...

I think we have `modify(...)` which sets `artifact.details  = details` later...

Reply:
Another option is `details = artifact.details` that's probably safer...

Note: I could be wrong here, maybe I misunderstood the flow...

Reply:
No, we don't store the `uploadId` we just created, we cancel the one that we just created and keep the old one and use the old one.

Reply:
I'm not sure what is causing the failure on travis, because it runs perfectly on my mac locally.  I need to debug the test a little more

Reply:
Correct, but we do make a `.modify` this would cause the `.modify` to be a  no-op.